packages/firestore/src/remote/remote_store.ts

```diff
@@ -136,82 +152,66 @@ export class RemoteStore implements TargetMetadataProvider {
-    return this.enableNetwork();
-  }
+  async start(): Promise<void> {
+    // Load any saved stream token from persistent storage
+    this.writeStream.lastStreamToken = await this.localStore.getLastStreamToken();
 
-  private isNetworkEnabled(): boolean {
-    assert(
-      (this.watchStream == null) === (this.writeStream == null),
-      'WatchStream and WriteStream should both be null or non-null'
-    );
-    return this.watchStream != null;
+    await this.enableNetwork();
   }
 
   /** Re-enables the network. Idempotent. */
-  enableNetwork(): Promise<void> {
-    if (this.isNetworkEnabled()) {
-      return Promise.resolve();
-    }
-
-    // Create new streams (but note they're not started yet).
-    this.watchStream = this.datastore.newPersistentWatchStream();
-    this.writeStream = this.datastore.newPersistentWriteStream();
-
-    // Load any saved stream token from persistent storage
-    return this.localStore.getLastStreamToken().then(token => {
-      this.writeStream.lastStreamToken = token;
+  async enableNetwork(): Promise<void> {
+    if (!this.networkEnabled) {
+      this.networkEnabled = true;
+      this.writeStream.lastStreamToken = await this.localStore.getLastStreamToken();
 
       if (this.shouldStartWatchStream()) {
         this.startWatchStream();
       } else {
         this.onlineStateTracker.set(OnlineState.Unknown);
       }
 
-      return this.fillWritePipeline(); // This may start the writeStream.
-    });
+      // This will start the write stream if necessary.
+      await this.fillWritePipeline();
+    }
   }
 
   /**
    * Temporarily disables the network. The network can be re-enabled using
    * enableNetwork().
    */
   async disableNetwork(): Promise<void> {
-    this.disableNetworkInternal();
+    await this.disableNetworkInternal();
+
     // Set the OnlineState to Offline so get()s return from cache, etc.
     this.onlineStateTracker.set(OnlineState.Offline);
   }
 
-  /**
-   * Disables the network, if it is currently enabled.
-   */
-  private disableNetworkInternal(): void {
-    if (this.isNetworkEnabled()) {
-      // NOTE: We're guaranteed not to get any further events from these streams (not even a close
-      // event).
-      this.watchStream.stop();
-      this.writeStream.stop();
+  private async disableNetworkInternal(): Promise<void> {
+    if (this.networkEnabled) {
+      this.networkEnabled = false;
 
-      this.cleanUpWatchStreamState();
+      this.writeStream.stop();
+      this.watchStream.stop();
 
-      log.debug(
-        LOG_TAG,
-        'Stopping write stream with ' +
-          this.writePipeline.length +
-          ' pending writes'
-      );
-      // TODO(mikelehen): We only actually need to clear the write pipeline if
-      // this is being called as part of handleUserChange(). Consider reworking.
-      this.writePipeline = [];
+      if (this.writePipeline.length > 0) {
+        log.debug(
+          LOG_TAG,
+          'Stopping write stream with ' +
```

Comment:
Nit: I'm guesstimating that this might fit on one line if you change it to use backticks:

```
`Stopping write stream with ${this.writePipeline.length} pending writes`
```

Reply:
Nope! Nice try though.